usaspending_api/agency/v2/views/object_class_list.py

```diff
@@ -0,0 +1,74 @@
+from django.db.models import Q, Sum, F
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any, List
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, ListMixin
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_simple_pagination_metadata
+from usaspending_api.references.models import ObjectClass
+
+
+class ObjectClassList(ListMixin, AgencyBase):
+    """
+    Obtain the list of object classes for a specific agency in a single
+    fiscal year based on whether or not that object class has ever
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/object_class.md"
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        results = list(self.get_object_class_list())
+        count = len(results)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        page_metadata = get_simple_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        page_metadata["count"] = count
```

Comment:
There is another pagination metadata function in generic_helper.py: `get_pagination_metadata()`

Up to you, @emily-brents, and @ebdabbs if the metadata object should be different (`total` instead of `count`, and adds `limit`)

Reply:
Indifferent to me personally. It appears we have at least three different ways we convey similar pagination. Happy to change it if we make the decision to do so.

Reply:
If we can be more consistent with `page_metadata` objects for other endpoints, that would be my preference.

Reply:
OK, thanks for your opinion. I agree

@sethstoudenmier @emily-brents I know it is late. Can each of you make the changes to your PRs?